src/openms/source/ANALYSIS/ID/FragmentIndex.cpp

```diff
@@ -252,6 +270,156 @@ namespace OpenMS
       return hits;
   }
 
+  void FragmentIndex::queryPeak(OpenMS::FragmentIndex::SpectrumMatchesTopN& candidates,
+                                const OpenMS::Peak1D& peak,
+                                std::pair<size_t, size_t> candidates_range,
+                                int16_t isotope_error,
+                                uint16_t precursor_charge)
+  {
+      for (uint16_t fragment_charge = 1; fragment_charge <= max_fragment_charge_; fragment_charge++)
+      {
+        for (auto hit : query(peak, candidates_range, fragment_charge))
+        {
+          // the following part is 1:1 from sage
+          size_t idx = hit.peptide_idx - candidates_range.first;
+
+          auto& source = candidates.hits_[idx];
+          if (source.num_matched_ == 0)
+          {
+            ++candidates.scored_candidates_;
+            source.precursor_charge_ = precursor_charge;
+            source.peptide_idx_ = hit.peptide_idx;
+            source.isotope_error_ = isotope_error;
+          }
+          ++source.num_matched_;
+          ++candidates.matched_peaks_;
+        }
+      }
+  }
+
+  void FragmentIndex::trimHits(OpenMS::FragmentIndex::SpectrumMatchesTopN& init_hits) const
+  {
+      if (max_processed_hits_ < init_hits.hits_.size())
+      {
+        std::partial_sort(init_hits.hits_.begin(), init_hits.hits_.begin() + max_processed_hits_, init_hits.hits_.end(), [](SpectrumMatch a, SpectrumMatch b){
+          return a.num_matched_ > b.num_matched_;
+        });
+
+        init_hits.hits_.resize(max_processed_hits_);
+      }
+      for(auto hit_iter = init_hits.hits_.begin(); hit_iter != init_hits.hits_.end();){
```

Comment:
also add space after for and put { into new line. Just some style convention we have ...